DSharpPlus.UnifiedCommands/Application/Internals/ApplicationHandler.cs

```diff
@@ -0,0 +1,148 @@
+            _module.Interaction = _interaction;
+            _module.Client = _client;
+            _module._handler = this;
+
+            foreach (Func<IServiceProvider, IApplicationCondition> conditionBuilder in _conditionBuilders)
+            {
+                IApplicationCondition condition = conditionBuilder(_scope.ServiceProvider);
+                Task<bool> task = condition.InvokeAsync(_interaction, _client);
+                bool shouldContinue = await task;
+
+                if (!shouldContinue)
+                {
+                    return;
+                }
+            }
+
+            try
+            {
+                if (!_data.ReturnsNothing)
+                {
+                    if (_data.IsAsync)
+                    {
+                        Task<IApplicationResult> task =
+                            (Task<IApplicationResult>)_data.Method.Invoke(_module, BindingFlags.OptionalParamBinding,
+                                null,
+                                _args, null)!;
+                        await TurnResultIntoActionAsync(await task);
+                    }
+                    else
+                    {
+                        IApplicationResult result = (IApplicationResult)_data.Method.Invoke(_module,
+                            BindingFlags.OptionalParamBinding, null, _args, null)!;
+                        await TurnResultIntoActionAsync(result);
+                    }
+                }
+                else
+                {
+                    if (_data.IsAsync)
+                    {
+                        Task task =
+                            (Task)_data.Method.Invoke(_module, BindingFlags.OptionalParamBinding, null, _args, null)!;
+                        await task;
+                    }
+                    else
+                    {
+                        _data.Method.Invoke(_module, BindingFlags.OptionalParamBinding, null, _args, null);
+                    }
+                }
+            }
+            catch (TargetInvocationException e)
+            {
+                // ExceptionDispatchInfo can be used to rethrow a error and perceive all the info in it.
+                // This is done here to rethrow the inner error and show where it is so people who don't read in depth their stacktrace doesn't blame us.
+                System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e.InnerException ?? e).Throw();
+                throw e.InnerException;
+            }
+            catch (AggregateException e)
+            {
+                System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e.InnerException ?? e).Throw();
+                throw e.InnerException;
+            }
+        }
+        catch (Exception e)
+        {
+            _client.Logger.LogError(
+                e,
+                "Exception was thrown while trying to execute command {MethodName}. Was thrown from ",
```

Comment:
Is this formatting right? looks as if there is something missing at the end. AFAIK the logger appends the exception which will be worded weird.:
```
Exception was thrown while trying to execute command {MethodName}. Was thrown from
System.Exception
....
```